### ISSUE TITLE
Error with GUID query variables

### ISSUE THREAD

**Issue by bugged84** [graphql-dotnet/graphql-dotnet#962]
## Description

I'm having an issue with GUID query variables.

### Steps to reproduce

```
public class MyQuery : ObjectGraphType
{
   public MyQuery(IAnimalRepository animalRepository)
   {
      Field<AnimalGraphType>(
         "animal"
       , arguments:
         new QueryArguments(
            new QueryArgument<NonNullGraphType<GuidGraphTypeAdapter>>
            {
               Name = "uuid"
            }
         )
       , resolve:
         context =>
            animalRepository
              .GetAsync(context.GetArgument<Guid>("uuid"))
              .Result
      );
   }
}

public class AnimalGraphType : ObjectGraphType<Animal>
{
   public AnimalGraphType()
   {
      Name = "Animal";

      Field("uuid", x => x.Guid, type: typeof(GuidGraphTypeAdapter));
      Field(x => x.Name, nullable: true);
   }
}

public class Animal
{
   public Guid Guid { get; set; }
   public string Name { get; set; }
}

public class GuidGraphTypeAdapter : GuidGraphType
{
   public GuidGraphTypeAdapter()
   {
      Name = "Uuid";
   }

   public override object ParseLiteral(IValue value)
   {
      switch (value)
      {
         case GuidValue guidValue:

            return guidValue.Value;
         case StringValue stringValue:

            return ParseValue(stringValue.Value);
         default:

            return null;
      }
   }
}
```

Request
```
{
	"query": "query($uuid: Uuid!) {
		animal(uuid: $uuid) {
			uuid
			name
		}
	}",
	"variables":{
		"uuid": "7F100A22-3BD2-E811-AF11-00155DFA2F51"
	}
}
```

### Expected result

No error.

### Actual result

```
{
    "errors": [
        {
            "message": "Cannot convert value to AST: 7f100a22-3bd2-e811-af11-00155dfa2f51",
            "extensions": {
                "code": "EXECUTION_ERROR"
            }
        }
    ]
}
```

**Comment by bugged84** [comment]
If I change my query to pass the argument inline rather than using a variable, then it works.

```
{
	"query": "mutation {
		updateAnimal(animal: { id: \"7f100a22-3bd2-e811-af11-00155dfa2f51\", name: \"Bones2\" }) {
			id,
			name
		}
	}"
}

So what is wrong with my original query using variables?
```

**Comment by bugged84** [comment]
@joemcbride any thoughts on this one?

**Comment by joemcbride** [comment]
Nothing that I can tell off-hand.  Can you provide a failing test?

**Comment by joemcbride** [comment]
How are you passing your variables to the executor?  Are you using `ToInputs`?

https://graphql-dotnet.github.io/docs/getting-started/variables

**Comment by joemcbride** [comment]
I think I just realized what your issue is - since your Scalar is converting the value to a `Guid`, that `Guid` needs to be converted to an `IValue`.  There is a `GuidValue` class which you are using in your `ParseLiteral` function.  You need to teach the `Schema` about that.  This is done manually in this pr https://github.com/graphql-dotnet/graphql-dotnet/pull/917 though it is not merged yet.

For you to do it, you will need to create an `IAstFromValueConverter` and register it with your Schema.

```csharp
schema.RegisterValueConverter(new GuidValueConverter());

    internal class GuidValueConverter : IAstFromValueConverter
    {
        public bool Matches(object value, IGraphType type)
        {
            return value is Guid;
        }

        public IValue Convert(object value, IGraphType type)
        {
            return new GuidValue((Guid)value);
        }
    }
```

**Comment by bugged84** [comment]
Yes, I'm passing variables to the executor by assigning the result of `ToInputs` to the  `ExecutionOptions.Inputs` property.

Registering the `GuidValueConverter` implementation you provided fixed the issue for me. Thanks for that.

**Comment by sungam3r** [comment]
Сhanges from #917 and other work on scalars are already merged into master. It is no longer necessary to manually register anything for the `Guid` type.

**Comment by joemcbride** [comment]
Thanks @sungam3r - closing this @bugged84

**Comment by ericcanadas** [comment]
Hi, th fix is published in 2.4.0 version ?

**Comment by sungam3r** [comment]
No, try the latest 3.0.0-preview.
